src/figury/AnimPanel.java

```diff
@@ -63,7 +62,12 @@ void animate() {
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		device.drawImage(image, 0, 0, null);
-		buffer.clearRect(0, 0, getWidth(), getHeight());
+		SwingUtilities.invokeLater(new Runnable(){
```

Comment:
tu nie trzeba. `actionPerformed` jest we właściwym wątku, a opóźnienie też nie jest potrzebne